src/Analyzers/CSharp/Analyzers/MakeStructMemberReadOnly/CSharpMakeStructMemberReadOnlyAnalyzer.cs

```diff
@@ -47,19 +57,38 @@ protected override void InitializeWorker(AnalysisContext context)
                         DeclaringSyntaxReferences: [var reference, ..],
                     } structType)
                 {
-                    return;
+                    option = null;
+                    return false;
                 }
 
                 var cancellationToken = context.CancellationToken;
                 var declaration = reference.GetSyntax(cancellationToken);
                 var options = context.GetCSharpAnalyzerOptions(declaration.SyntaxTree);
-                var option = options.PreferReadOnlyStructMember;
+                option = options.PreferReadOnlyStructMember;
                 if (!option.Value)
-                    return;
+                    return false;
 
-                context.RegisterOperationBlockAction(
-                    context => AnalyzeBlock(context, option.Notification.Severity));
-            }, SymbolKind.NamedType);
+                // Skip analysis if the analysis filter span does not contain the primary location where we would report a diagnostic.
+                if (context.FilterSpan.HasValue)
+                {
+                    Contract.ThrowIfNull(context.FilterTree);
+                    var shouldAnalyze = false;
+                    foreach (var member in structType.GetMembers())
+                    {
+                        var location = GetDiagnosticLocation(member, cancellationToken, out _);
+                        if (location != null && context.ShouldAnalyzeLocation(location))
+                        {
+                            shouldAnalyze = true;
+                            break;
+                        }
```

Comment:
```suggestion
                        if (location != null && context.ShouldAnalyzeLocation(location))
                            return true;
```

Reply:
Same comment as previous - current code style allows us to add more bailout checks in the future without having to restructure this code.